temporal-sdk/src/main/java/io/temporal/internal/statemachines/WorkflowStateMachines.java

```diff
@@ -416,6 +513,22 @@ public List<Command> takeCommands() {
     return result;
   }
 
+  public void sendMessage(Message message) {
+    checkEventLoopExecuting();
+    if (!isReplaying()) {
+      messageOutbox.add(message);
+    }
+  }
+
+  public List<Message> takeMessages() {
```

Comment:
Minor: Probably cheaper to grab a temp ref to messageOutbox, reassign messageOutbox = new ArrayList<>() and then return the temp ref here.